src/test/java/com/epam/selenium/pages/tests/CheapFlightsTest.java

```diff
@@ -0,0 +1,120 @@
+package com.epam.selenium.pages.tests;
+
+import com.epam.selenium.pages.desktop.EmptyHomePage;
+import com.epam.selenium.pages.desktop.FirstFlightSearchPage;
+import com.epam.selenium.pages.desktop.PrefilledHomePage;
+import com.epam.selenium.pages.desktop.SecondFlightSearchPage;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.firefox.FirefoxOptions;
+import org.openqa.selenium.remote.RemoteWebDriver;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Parameters;
+import org.testng.annotations.Test;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.concurrent.TimeUnit;
+
+public class CheapFlightsTest {
+
+    private WebDriver driver;
+
+    private By logoXpath = By.xpath("//div[contains(@class, 'logo')]//a[@href='/']");
+    private By originFieldName = By.name("origin");
+
+    private String originFieldAttribute = "value";
+    private String originFieldValue = "";
+
+    String hubUrl = "http://10.6.182.106:4444/wd/hub";
+
+
+    @BeforeClass
+    @Parameters({"browser"})
+    public void launchBrowser() {
+        // System.setProperty("webdriver.gecko.driver", "./src/main/resources/geckodriver");
+        FirefoxOptions options = new FirefoxOptions();
+        try {
+            driver = new RemoteWebDriver(new URL(hubUrl), options);
+        } catch (MalformedURLException e) {
+            e.printStackTrace();
```

Comment:
Provided malformed URL there is not reasons to to continue test steps.

Handle it or abort right there